### ISSUE TITLE
function declaration should goes to the top level

### ISSUE THREAD

**Issue by curimit** [BYVoid/continuation#30]
```javascript
test();
setTimeout(cont(), 100);
function test() {
  console.log('test');
}
```
Which compiles to
```javascript
test();
setTimeout(function (arguments) {
  function test() {
    console.log('test');
  }
}.bind(this, arguments), 100);
```
This is incorrect.

**Comment by BYVoid** [comment]
Thanks for noticing this. I just found this works
```js
test();
function test() {
  console.log("a");
}
```

while this doesn't work:
```js
test();
var test = function() {
  console.log("a");
}
```

**Comment by curimit** [comment]
The problem here is that, if you warp `function test()` in callbacks, neither of these two methods will working
```javascript
test();
setTimeout(function() {
  function test() {
    console.log("a");
  }
}, 0);
```

**Comment by curimit** [comment]
Fixed: 543ac7e
